docs/building/building-environment.md

```diff
@@ -0,0 +1,153 @@
+to create and use docker controlled environments to build & test `tracee-ebpf`
+and `tracee-rules`.
+
+Two different environments are maintained for building tracee:
+
+* Alpine
+* Ubuntu
+
+The reason for that is that **Alpine Linux** is based in the
+[musl](https://en.wikipedia.org/wiki/Musl) C standard library, while the
+**Ubuntu Linux** uses [glibc](https://en.wikipedia.org/wiki/Glibc). By
+supporting both building environments we can always be sure that the project
+builds (and executes) correctly in both environments.
+
+Be aware: local created containers, called `alpine-tracee-make` or
+`ubuntu-tracee-make`, share the host source code directory. This means that, if
+you build tracee binaries using `alpine` distribution, binaries `tracee-ebpf`
+and `tracee-rules` might not be compatible to the Linux distribution from your
+host OS.
+
+### CREATING A BUILDER ENVIRONMENT
+
+To create an **alpine-tracee-make** container:
+
+  ```
+  $ make -f builder/Makefile.tracee-make alpine-prepare
+  ```
+
+To create an **ubuntu-tracee-make** container:
+
+  ```
+  $ make -f builder/Makefile.tracee-make ubuntu-prepare
+  ```
+
+### EXECUTING A BUILDER ENVIRONMENT
+
+To execute an **alpine-tracee-make** shell:
+
+  ```
+  $ make -f builder/Makefile.tracee-make alpine-shell
+  ```
+
+To execute an **ubuntu-tracee-make** shell:
+
+  ```
+  $ make -f builder/Makefile.tracee-make ubuntu-shell
+  ```
+
+### USING BUILD ENV AS "MAKE" REPLACEMENT
+
+Instead of executing a builder shell, you may use `alpine-tracee-make`, or
+`ubuntu-tracee-make`, as a replacement for the `make` command:
+
+```
+$ make -f builder/Makefile.tracee-make ubuntu-prepare
+$ make -f builder/Makefile.tracee-make ubuntu-make ARG="help"
+$ make -f builder/Makefile.tracee-make ubuntu-make ARG="clean"
+$ make -f builder/Makefile.tracee-make ubuntu-make ARG="bpf-core"
+$ make -f builder/Makefile.tracee-make ubuntu-make ARG="tracee-ebpf"
+$ make -f builder/Makefile.tracee-make ubuntu-make ARG="all"
+```
+
+And, after the compilation, run the commands directly in your host
+
+```
+$ sudo ./dist/tracee-ebpf \
+	-o option:parse-arguments \
+	--trace comm=bash \
+	--trace follow \
+	--trace event!='sched*'
+```
+
+if the generated binary is compatible (depending on glibc version, for example).
+
+If you don't want to depend on host's libraries versions, or if you are using
+the `alpine-tracee-make` container as a replacement for `make`, and your host
+is not an **Alpine Linux**, then you may set STATIC=1 variable so you can run
+compiled binaries in your host:
+
+```
+$ make -f builder/Makefile.tracee-make alpine-prepare
+$ make -f builder/Makefile.tracee-make alpine-make ARG="help"
+$ STATIC=1 make -f builder/Makefile.tracee-make alpine-make ARG="all"
+```
+
+and execute the static binary from your host:
+
+```
+$ ldd dist/tracee-ebpf
+	not a dynamic executable
+```
+
+> **Warning**: compiling `tracee-rules` with STATIC=1 won't allow you to use
```

Comment:
You can use the following mkdocs syntax to add the Warning box:

<pre>
!!! warning
    Here is your warning
<pre>

Reply:
Yep, I didn't because github markdown does not support it.

But thats a thing for us to go through in the final docs review and make sure we have same formatting in all markdowns.